sympy/physics/control/tests/test_lti.py

```diff
@@ -138,13 +139,11 @@ def test_TransferFunction_construction():
 
     # no pole-zero cancellation on its own.
     tf4 = TransferFunction((s + 3)*(s - 1), (s - 1)*(s + 5))
-    assert tf4.den == (s**2 + 4*s - 5)
-    assert tf4.args == (s**2 + 2*s - 3, s**2 + 4*s - 5)
+    assert tf4.den == (s - 1)*(s + 5)
+    assert tf4.args == ((s + 3)*(s - 1), (s - 1)*(s + 5))
 
-    # purely symbolic polynomials.
-    tf5 = TransferFunction(p, p)
-    assert tf5.num == tf5.den == p
-    assert tf5.args == (p, p)
+    tf5 = TransferFunction(s - 1, 4 - p, s)
+    assert tf5.args == (s - 1, 4 - p)
 
     tf6 = TransferFunction(5, 6)
```

Comment:
How about this? 
1. The user doesn't have to specify the `var` when the transfer function is like this -  `TransferFunction(s**4 + s**3 - 2*s, s**4 - 1)` (`s` being the default variable, as it is most commonly used.)
2. `var` is also not needed when both the numerator and the denominator are numbers. e.g. `TransferFunction(5, 6)`.
3. In all other cases like `TransferFunction(s - 1, 4 - p, s)` and `TransferFunction(p + 3, p**2 - 9, p)` we need to specify the `var` as the third argument.

I think passing `s` as the third argument every time is unnecessary.

Reply:
I think we should pass the variable explicitly in all cases. The variable should also be always accessible from `.args[2]` if the user needs the variable. Here is another example showing why this is needed in user code;

```
tf1 = TransferFunction(5, 6, s)
tf2 = TransferFunction(5, 6, z)
tf1 == tf2
tf1*tf2
```
That should raise an error right?

Should this?
```
tf1 = TransferFunction(5, 6)
tf2 = TransferFunction(5, 6, s)
tf1 == tf2
tf1*tf2
```
It is ambiguous now, because there is no explicitness about what the variable is for tf1. A transfer function is not a transfer function if it isn't a function of a complex variable.

I think the best decision will be making it explicit and always required.

Reply:
You can add a property:
```
    @property 
    def variable(self):
        return self.args[2]
```
Then users can use `tf.variable` to access it. Users should be discouraged from accessing args directly in case the layout of the args ever needs to change.

Using properties rather than direct args access also makes it possible to share code between subclasses that have different args layouts.

Reply:
I agree with @oscarbenjamin on this. Please add an attribute for the variable.

Reply:
> I think the best decision will be making it explicit and always required.

Thanks for pointing this! 



> I agree with @oscarbenjamin on this. Please add an attribute for the variable.

That's good. I will do that. Similarly, we can also add `num` and `den` as a `property`.
``` python
@property
def num(self):
    return self._num

@property
def den(self):
    return self._den
```